### PR TITLE
Maximum price calculation additional info field

### PR DESCRIPTION
**Pull request by ranta**
# Hitas Pull Request

# Description

- Add `additional_field` to confirmed and unconfirmed maximum price PDFs.
- For confirmed maximum price, it's entered when creating a calculation.
- For unconfirmed (estimate) it's entered before downloading the PDF.
  - The old button now opens a modal, with 'Lisätieto'-field. The value is sent to the API and the text is inserted in the PDF when generating it.

## Pull request checklist

Check the boxes for each DoD items that has been completed:

- **Frontend**
    - [x] Changes have been tested
- **Backend**
    - [x] Changes have been tested
    - [x] Automatic tests has been added
    - [x] OpenAPI definitions have been updated
    - [x] Database migrations will work in test environment

## Test plan

- Create a confirmed price calculation and make sure that the json on the pdf contains `additional_info` and the text you entered
- Download an unconfirmed price PDF, and enter some text in the model, make sure the PDF contains the text you entered.

## Tickets

This pull request resolves all or part of the following ticket(s): HT-329

### CODE REVIEW

File: backend/openapi.yaml [hunk 1]
```diff
@@ -926,6 +930,19 @@ paths:
           schema:
             type: string
             example: b477389cb3514fb1b444052a39bfb65d
+      requestBody:
+        required: true
```
Comment (toikarin-vincit) [inline backend/openapi.yaml:934]: This could be optional?

File: backend/openapi.yaml [hunk 2]
```diff
@@ -926,6 +930,19 @@ paths:
...
+        content:
+          application/json:
+            schema:
+              description: TBD
```
Comment (toikarin-vincit) [inline backend/openapi.yaml:938]: define :)

File: backend/openapi.yaml [hunk 3]
```diff
@@ -926,6 +930,19 @@ paths:
...
+              type: object
+              additionalProperties: false
+              properties:
+                additional_info:
+                  description: Additional information to be displayed on the generated PDF
```
Comment (toikarin-vincit) [inline backend/openapi.yaml:943]: `additional_info` is the correct translation but I'm thinking is a `notes` a better term (we already use that). `additional_info` sounds more like an object than a string. Other option would be to add "Additional **textual** information" to the description

File: frontend/src/features/apartment/ApartmentDetailsPage.tsx [hunk 4]
```diff
@@ -76,16 +102,144 @@ const ConfirmedPrice = ({confirmed}) => {
...
+                    variant="secondary"
+                    theme="black"
+                    size="small"
+                >
+                    Sulje
+                </Button>
+                <Button
+                    theme="black"
+                    size="small"
+                    variant="secondary"
```
Comment (vincit-matu) [inline frontend/src/features/apartment/ApartmentDetailsPage.tsx:149]: This button shouldn't have the `variant="secondary"` attribute, as it's the main action button in the modal.

**Review by vincit-matu — CHANGES_REQUESTED** [review]

File: frontend/src/features/apartment/ApartmentDetailsPage.tsx (hunk 4 above)
Comment (ranta) [reply frontend/src/features/apartment/ApartmentDetailsPage.tsx]: Fixed :)

**Review by toikarin-vincit — APPROVED** [review]
LGTM :rocket:

**Review by vincit-matu — APPROVED** [review]

**Pull request merged**